### PR TITLE
Change message category

### PR DESCRIPTION
**Pull request by insacuri**
# What and why?

# How to test?

# Trello

### CODE REVIEW

File: secure_message/repository/modifier.py [hunk 1]
```diff
@@ -133,7 +133,58 @@ def _mark_read(message, user):
...
+        bound_logger = logger.bind(conversation_id=conversation.id)
+
+        for field in ['category']:
+            # Looks a bit awkward but getattr/setattr lets us loop over all the fields as we can't access fields
+            # in the object like a dictionary.
+
+            if request_data.get(field) and request_data.get(field) != getattr(conversation, field):
+                setattr(conversation, field, request_data[field])
+            else:
+                logger.info("Value is the same as what is in the database, leaving unchanged", field=field)
```
Comment (mfcrocker) [inline secure_message/repository/modifier.py:154]: This is possibly a little too chatty

File: secure_message/repository/modifier.py [hunk 2]
```diff
@@ -133,7 +133,58 @@ def _mark_read(message, user):
...
+        try:
+            db.session.commit()
+        except SQLAlchemyError:
+            db.session.rollback()
+            bound_logger.exception("Database error occurred while opening conversation")
```
Comment (mfcrocker) [inline secure_message/repository/modifier.py:159]: Are we swallowing the exception here? I'd rather we had details of the exception in the logs

File: secure_message/repository/modifier.py [hunk 3]
```diff
@@ -133,7 +133,58 @@ def _mark_read(message, user):
...
+                # Currently not possible to clear a value in a field.
+                if request_data.get(field) and request_data.get(field) != getattr(message, field):
+                    setattr(message, field, request_data[field])
+                else:
+                    logger.info("Value is the same as what is in the database, leaving unchanged", field=field)
+
+            db.session.commit()
+        except SQLAlchemyError:
+            db.session.rollback()
+            bound_logger.exception("Database error occurred while opening conversation")
```
Comment (mfcrocker) [inline secure_message/repository/modifier.py:183]: As above re: exceptions

File: secure_message/repository/modifier.py [hunk 4]
```diff
@@ -133,7 +133,58 @@ def _mark_read(message, user):
...
+        :param message: An dict containing the currently saved database data
+        """
+        bound_logger = logger.bind(message_id=message.msg_id)
+        try:
+            for field in ['survey_id', 'case_id', 'business_id', 'exercise_id']:
+                # Currently not possible to clear a value in a field.
+                if request_data.get(field) and request_data.get(field) != getattr(message, field):
+                    setattr(message, field, request_data[field])
+                else:
+                    logger.info("Value is the same as what is in the database, leaving unchanged", field=field)
```
Comment (mfcrocker) [inline secure_message/repository/modifier.py:178]: As above re: chatty

File: secure_message/repository/modifier.py [hunk 5]
```diff
@@ -149,14 +200,14 @@ def close_conversation(metadata, user):
             db.session.commit()
         except SQLAlchemyError:
             db.session.rollback()
-            bound_logger.exception("Database error occured while closing conversation")
-            raise InternalServerError(description="Database error occured while closing conversation")
+            bound_logger.exception("Database error occurred while closing conversation")
```
Comment (mfcrocker) [inline secure_message/repository/modifier.py:203]: As above

File: secure_message/repository/retriever.py [hunk 6]
```diff
@@ -202,8 +202,28 @@ def _retrieve_internal_thread_list(request_args, user):
...
+        """
+        logger.info("Retrieving message", message_id=message_id)
+        try:
+            result = SecureMessage.query.filter_by(msg_id=message_id).one()
+            if result is None:
+                logger.info('Message ID not found', message_id=message_id)
+                raise NotFound(description=f"Message with msg_id '{message_id}' does not exist")
+        except SQLAlchemyError:
+            logger.exception('Error retrieving message from database')
+            raise InternalServerError(description="Error retrieving message from database")
```
Comment (mfcrocker) [inline secure_message/repository/retriever.py:220]: As above

**Review by mfcrocker — APPROVED** [review]
LGTM, just a couple of points on logging to consider

File: secure_message/repository/modifier.py (hunk 2 above)
Comment (insacuri) [reply secure_message/repository/modifier.py:159]: logger.exception puts the stack trace in the logs for free

File: secure_message/repository/modifier.py (hunk 1 above)
Comment (insacuri) [reply secure_message/repository/modifier.py:154]: Yeah, I think I agree.  Gonna remove it.

File: secure_message/repository/modifier.py (hunk 3 above)
Comment (insacuri) [reply secure_message/repository/modifier.py:183]: As above re: stack trace is included ^^

File: secure_message/repository/modifier.py (hunk 4 above)
Comment (insacuri) [reply secure_message/repository/modifier.py:178]: Removed

**Pull request merged**